Sources/Java2SwiftLib/JavaTranslator+Validation.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift.org project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+package extension JavaTranslator {
+  struct SwiftTypeName: Hashable {
+    let swiftType: String
+    let swiftModule: String?
+
+    package init(swiftType: String, swiftModule: String?) {
+      self.swiftType = swiftType
+      self.swiftModule = swiftModule
+    }
+  }
+
+  struct SwiftToJavaMapping: Equatable {
+    let swiftType: SwiftTypeName
+    let javaTypes: [String]
+
+    package init(swiftType: SwiftTypeName, javaTypes: [String]) {
+      self.swiftType = swiftType
+      self.javaTypes = javaTypes
+    }
+  }
+
+  enum ValidationError: Error, CustomStringConvertible {
+    case multipleClassesMappedToSameName(swiftToJavaMapping: [SwiftToJavaMapping])
+
+    package var description: String {
+      switch self {
+      case .multipleClassesMappedToSameName(let swiftToJavaMapping):
+              """
+              The following Java classes were mapped to the same Swift type:
+                \(swiftToJavaMapping.map { mapping in
+                  "Swift Type: \(mapping.swiftType.swiftModule ?? "").\(mapping.swiftType.swiftType), Java Types: \(mapping.javaTypes.sorted().joined(separator: ", "))"
```

Comment:
```suggestion
                  "Swift Type: '\(mapping.swiftType.swiftModule ?? "")'.\(mapping.swiftType.swiftType), Java Types: \(mapping.javaTypes.sorted().map({ "'\($0)'" })joined(separator: ", "))"
```

I'd split out the creating of the names into separate lines, this is a bit busy inside the string interpolation

Reply:
updated and made a helper function to split it out even more